modules/exploits/example.rb

```diff
@@ -0,0 +1,84 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+###
+#
+# This exploit sample shows how an exploit module could be written to exploit
+# a bug in an arbitrary TCP server.
+#
+###
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  #
+  # This exploit affects TCP servers, so we use the TCP client mixin.
+  #
+  include Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
```

Comment:
Note that 'update_info' is usually best!

We should talk a little about 'merge_info' and when you should use it.